src/test/javascript/grouping.controller.test.js

```diff
@@ -2529,41 +2529,122 @@ describe("GroupingController", () => {
+
+        it("should set $scope.pageSelected to false if checkboxClicked is \"page\"", () => {
             scope.pageSelected = true;
             scope.toggleCheckAllSelection("Include", "page");
             expect(scope.pageSelected).toBeFalse();
         });
-        it("should set the membersInCheckboxList.uhUuid to the value of scope.pageSelected", () => {
+
+        it("should call checkMainSelectAllCheckbox", () => {
+            spyOn(scope, "checkMainSelectAllCheckbox").and.callThrough();
+            scope.toggleCheckAllSelection("Include", "page");
+            expect(scope.checkMainSelectAllCheckbox).toHaveBeenCalled();
+        });
+
+        it("should set $scope.pageSelected and $scope.toolTip to be check's values", () => {
+            scope.toggleCheckAllSelection("Include", "page");
+            expect(scope.pageSelected).toBeTrue();
+            expect(scope.showToolIcon).toBeFalse();
+        });
+
+        it("should call updateCheckboxEventListeners", () => {
+            spyOn(scope, "updateCheckboxEventListeners").and.callThrough();
+            scope.toggleCheckAllSelection("Include", "other");
+            expect(scope.updateCheckboxEventListeners).toHaveBeenCalled();
+        });
+
+        it("should set the membersInCheckboxList.uhUuid to the value of $scope.pageSelected", () => {
             scope.pageSelected = false;
             scope.toggleCheckAllSelection("Include", "page");
             expect(scope.membersInCheckboxList[((scope.pagedItemsInclude[0][0]).uhUuid)]).toEqual(true);
         });
-        it("should set the membersInCheckboxList.uhUuid to the value of scope.pageSelected", () => {
+
+        it("should set the membersInCheckboxList.uhUuid to the value of $scope.pageSelected", () => {
             scope.pageSelected = true;
             scope.toggleCheckAllSelection("Include", "page");
             expect(scope.membersInCheckboxList[((scope.pagedItemsInclude[0][0]).uhUuid)]).toEqual(false);
         });
+
+        it("should change paginationPageChange from true to false", () => {
+            scope.paginationPageChange = true;
+            scope.toggleCheckAllSelection("Include", "other");
+            expect(scope.paginationPageChange).toBeFalse();
+        });
+    });
+
+    describe("checkMainSelectAllCheckbox", () => {
+        beforeEach(() => {
+            scope.testItem = {
+                allItems: scope.groupingInclude,
+                pageItems: scope.pagedItemsInclude,
+                pageNumber: scope.currentPageInclude
+            };
+        });
+
+        it("should return the original check values", () => {
+            scope.testItem.pageItems = [];
+            const testCheck = scope.checkMainSelectAllCheckbox(scope.testItem);
+            expect(testCheck.page).toBeTrue();
+            expect(testCheck.tooltip).toBeFalse();
+        });
+
+        it("should uncheck the Select All checkbox if there are any unchecked checkboxes", () => {
```

Comment:
There is this test that manually writes one false value in the `scope.membersInCheckboxList` which is the equivalent of unchecking a box. I'm not familiar with Selenium testing but think that should be a different ticket since this is for our Jasmine tests.